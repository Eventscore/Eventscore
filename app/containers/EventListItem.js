import React , { Component } from 'react';
import ReactNative from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ActionCreators } from '../actions';
import { Actions, ActionConst } from 'react-native-router-flux';

const {
  Image,
  ListView,
  TouchableHighlight,
  StyleSheet,
  Text,
  View,
} = ReactNative;

class EventListItem extends Component {
  constructor() {
    super();
    this.state = {
    };
  }

  pressListItem() {
    {Actions.eventview({type: ActionConst.PUSH})};
    this.props.changeCurrEvent(this.props.event);
  }

  componentWillMount() {
  }

  render() {
    const start = new Date(this.props.event.start);
    const startArray = start.toString() === 'Invalid Date' ? [] : start.toString().split(' '); 
    const day = startArray[0] || 'TBD';
    const date = startArray.slice(1, 3).join(' ') || 'TBD';
    
    let timeString = startArray[4] || ''; // convert to array
    let time = timeString.split(':');
    // fetch
    let hours = Number(time[0]);
    let minutes = Number(time[1]);

    // calculate
    let timeValue;

    if (hours > 0 && hours <= 12) {
      timeValue= "" + hours;
    } else if (hours > 12) {
      timeValue= "" + (hours - 12);
    } else if (hours === 0) {
      timeValue= "12";
    }
     
    timeValue += (minutes < 10) ? ":0" + minutes : ":" + minutes;  // get minutes
    timeValue += (hours >= 12) ? "PM" : "AM";  // get AM/PM

    return (
      <TouchableHighlight onPress={() => this.pressListItem()}>
      <View style={styles.container}>
        <View style={styles.dateBox}>
          <Text style={styles.weekday}>{day}</Text>
          <Text style={styles.date}>{date}</Text>
        </View>
        <View style={styles.eventInfo}>
          <Text style={styles.artist}>
            {this.props.event.artists[0] ? this.props.event.artists[0].name : 'Rum Ham'}
          </Text>
          <Text style={styles.headline}>{this.props.event.name}</Text>
          <Text style={styles.timeVenue}>
            {timeValue} @ {this.props.event.venue ? this.props.event.venue : 'Wrigley Field'}
          </Text>
          <Text style={styles.location}>
            {this.props.event.city ? this.props.event.city : 'Chicago'}, {this.props.event.state ? this.props.event.state : 'IL'}
          </Text>
        </View>
        <View style={styles.scoreBox}>
          <Text style={styles.score}>Score:</Text>
          <Text style={styles.scoreNumber}>
            {this.props.event.artists[0] && this.props.event.artists[0].spotify.popularity ? this.props.event.artists[0].spotify.popularity : 69}
          </Text>
        </View>
      </View>
      </TouchableHighlight>
    );
          // {this.props.event.artists.map((artist) => {
          //   return (<Text key={artist.name}>{artist.name}</Text>)
          // })}
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#EEE',
    flex: 1,
    justifyContent: 'space-around',
    // height: 80,
    // borderBottomWidth: 0.3,
    padding: 3,
    margin: 1.5,
    // marginBottom: 3
  },
  dateBox:{
    display: 'flex',
    justifyContent: 'center',
    flex: 0.5
  },
  eventInfo:{
    display: 'flex',
    justifyContent: 'center',
    flex: 1
  },
  scoreBox:{
    display: 'flex',
    justifyContent: 'center',
    flex: 0.5
  },
  weekday:{
    textAlign: 'center',
    fontSize: 18
  },
  date:{
    textAlign: 'center',
  },
  artist:{
    textAlign: 'center',
  },
  headline:{
    textAlign: 'center',
    // flex: 1,
    fontSize: 18
  },
  timeVenue:{
    textAlign: 'center',
    fontSize: 12
  },
  location:{
    textAlign: 'center',
    fontSize: 12
  },
  score:{
    textAlign: 'center',
  },
  scoreNumber:{
    textAlign: 'center',
    fontSize: 18
  }
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators(ActionCreators, dispatch);
}

// export default EventListItem;
export default connect(({routes, loginReducers, eventsReducers, addCountExample}) => { return {routes, loginReducers, eventsReducers, addCountExample} }, mapDispatchToProps)(EventListItem);