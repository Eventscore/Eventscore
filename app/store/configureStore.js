//Uncertain if this will work
// import { createStore, applyMiddleware, combineReduxers, compose } from 'redux';
// import thunkMiddleware from 'redux-thunk';
// import reducers from '../reducers';
// import { createLogger } from 'redux-logger';

// //this logs only in development mode
// const loggerMiddleware = createLogger( { predicate: (getState, action) => __DEV__ });

// export default function configureStore() {
//   const enhancers = compose(
//     applyMiddleware(
//       thunkMiddleware,
//       loggerMiddleware,
//     ),
//   );

//   const store = createStore(reducers, enhancers);

//   if(module.hot) {
//     module.hot.accept( () => {
//       const nextRootReducer = require('../reducers/index').default;
//       store.replaceReducer(nextRootReducer);
//     });
//   }
//   return store;
// }

import { createStore, applyMiddleware, compose } from 'redux';
import reducers from '../reducers/index';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';

const loggerMiddleware = createLogger({ predicate: (getState, action) => __DEV__ })

export default function configureStore (initialState) {
  // const middleware = [thunkMiddleware, loggerMiddleware];
  const enhancer = compose(
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware
    )
  )
  const store = createStore(reducers, initialState, enhancer);
  // const store = compose(
  //   applyMiddleware(...middleware))(createStore)(reducers);

  // if (module.hot) {
  //   module.hot.accept(() => {
  //     const nextRootReducer = require('../reducers/index').default
  //     store.replaceReducer(nextRootReducer)
  //   })
  // }
  return store;
}

