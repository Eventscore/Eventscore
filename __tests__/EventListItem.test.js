import 'react-native';
import React from 'react';
import EventListItem from '../app/containers/EventListItem';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const initialState = {
  event: {
    artists: [{
      name: 'Jon Snow'
    }]
  }
};

const mockStore = configureStore(initialState);
const store = mockStore(initialState);
console.log('store: ', store);

xit('EventListItem renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><EventListItem /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});