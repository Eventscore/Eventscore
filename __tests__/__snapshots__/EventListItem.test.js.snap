// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListItem renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "display": "flex",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 150,
            "resizeMode": "cover",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.6)",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,.6)",
              "color": "#FFF",
              "fontSize": 16,
              "paddingLeft": 5,
              "position": "absolute",
              "textAlign": "left",
              "top": "55%",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFF5",
              "fontSize": 14,
              "paddingLeft": 5,
              "position": "absolute",
              "textAlign": "left",
              "top": "65%",
            }
          }
        >
          12:NaNAM
           @ 
          Undefined
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEE",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "margin": 1.5,
            "padding": 3,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-around",
              "padding": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            resizeMode="contain"
            style={
              Array [
                Object {
                  "color": "#7a7b7c",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#7a7b7c",
                "fontSize": 13,
              }
            }
          >
            TBD
            , 
            TBD
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-around",
              "padding": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            resizeMode="contain"
            style={
              Array [
                Object {
                  "color": "#7a7b7c",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#7a7b7c",
                "fontSize": 13,
              }
            }
          >
            N/A
            , 
            N/A
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "flex": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-around",
              "padding": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#7a7b7c",
                "fontSize": 13,
                "textAlign": "center",
              }
            }
          >
            Score:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#7a7b7c",
                "fontSize": 18,
              }
            }
          >
            69
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
